24th-App-Team-1-iOS/Feature/MessageFeature/Sources/Presentation/ViewControllers/Home/MessageHomeViewController.swift

```diff
@@ -0,0 +1,185 @@
+            $0.height.equalTo(433)
+        }
+    }
+    
+    private func animateBanner() {
+
+    }
+    
+    public override func setupAttributes() {
+        super.setupAttributes()
+        bottomInfoLabel.do {
+            $0.textColor = DesignSystemAsset.Colors.gray200.color
+        }
+    }
+    
+    public override func bind(reactor: Reactor) {
+        super.bind(reactor: reactor)
+        bindState(reactor: reactor)
+        bindAction(reacotr: reactor)
+        }
+}
+
+extension MessageHomeViewController {
+    private func bindState(reactor: MessageHomeViewReactor) {
+        
+        // 메시지 전송 가능 시간이 변경 및 타이머가 변경될 때마다 UI 업데이트
+        Observable.combineLatest(
+            reactor.pulse(\.$messageAvailability),
+            reactor.state.compactMap { $0.remainingTime }
+        )
+        .distinctUntilChanged { old, new in
+            let (oldAvail, oldTime) = old
+            let (newAvail, newTime) = new
+            // messageAvailabilityTime과 remainingTime 둘 다 동일하면 true
+            return oldAvail.messageAvailabilityTime == newAvail.messageAvailabilityTime
+                && oldTime == newTime
+        }
+        .observe(on: MainScheduler.instance)
+        .bind(with: self) { this, combined in
+            let (availability, remainingTime) = combined
+
+            let newHeight = MessageConstants.reservedMessageViewHeight(for: availability.messageAvailabilityTime)
+            this.messageTimeView.snp.updateConstraints { make in
+                make.height.equalTo(newHeight)
+            }
+            UIView.animate(withDuration: 0.3) {
+                this.view.layoutIfNeeded()
+            }
+            
+            this.bottomInfoLabel.isHidden = (availability.messageAvailabilityTime != .waitTime)
+            ? false : true
+            
+            this.bottomInfoLabel.text = availability.messageAvailabilityTime == .postableTime
+            ? String.MessageTexts.postableTimeIntroText : String.MessageTexts.waitTimeIntroText
+            
+            this.messageTimeView.configureMessageCountView(
+                leftTime: remainingTime,
+                availableSend: availability.messageAvailabilityTime
+            )
+        }
+        .disposed(by: disposeBag)
+        
+        globalState.event
+            .observe(on: MainScheduler.asyncInstance)
+            .bind(with: self) {  this, type in
+                switch type {
+                case .showToast(let message, let type):
+                    this.showWSToast(image: type, message: message)
+                default:
+                    break
+                }
+            }
+            .disposed(by: disposeBag)
+        
+        
+    }
+    
+    private func bindAction(reacotr: MessageHomeViewReactor) {
+        reacotr.action.onNext(.viewWillAppear)
+        self.rx.viewWillAppear
+            .bind { _ in
+                reacotr.action.onNext(.viewWillAppear)
+            }
+            .disposed(by: disposeBag)
```

Comment:
```suggestion
        self.rx.viewWillAppear
            .map { _ in Reactor.Action.viewWillAppear }
            .bind(to: reactor.action)
            .disposed(by: disposeBag)
```
- 위와 같은 형식으로 하는건 어떤가용??